File: app/lib/models/appointment.dart
class Appointment {
  final String id;
  final String uuid;
  final String studentId;
  final String counselorId;
  final DateTime appointmentDate;
  final String status;
  final String reason;
  final String? notes;
  final String locationType;
  final String? location;
  final Map<String, dynamic>? counselor;
  final Map<String, dynamic>? student;

  Appointment({
    required this.id,
    required this.uuid,
    required this.studentId,
    required this.counselorId,
    required this.appointmentDate,
    required this.status,
    required this.reason,
    this.notes,
    required this.locationType,
    this.location,
    this.counselor,
    this.student,
  });

  factory Appointment.fromJson(Map<String, dynamic> json) {
    return Appointment(
      id: json['id'].toString(),
      uuid: json['uuid'],
      studentId: json['student_id'].toString(),
      counselorId: json['counselor_id'].toString(),
      appointmentDate: DateTime.parse(json['appointment_date']),
      status: json['status'],
      reason: json['reason'],
      notes: json['notes'],
      locationType: json['location_type'] ?? 'online',
      location: json['location'],
      counselor: json['counselor'],
      student: json['student'],
    );
  }
} 

File: backend/app/Models/Appointment.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;
use Illuminate\Support\Str;
use Illuminate\Database\Eloquent\Concerns\HasUuids;

class Appointment extends Model
{
    use HasFactory, SoftDeletes, HasUuids;

    protected $fillable = [
        'student_id',
        'counselor_id',
        'appointment_date',
        'reason',
        'location_type',
        'location',
        'status',
    ];

    protected $casts = [
        'appointment_date' => 'datetime',
        'created_at' => 'datetime',
        'updated_at' => 'datetime'
    ];

    /**
     * Get the route key for the model.
     */
    public function getRouteKeyName(): string
    {
        return 'uuid';
    }

    /**
     * Get the columns that should receive a unique identifier.
     */
    public function uniqueIds(): array
    {
        return ['uuid'];
    }

    public function student()
    {
        return $this->belongsTo(User::class, 'student_id');
    }

    public function counselor()
    {
        return $this->belongsTo(User::class, 'counselor_id');
    }
} 